phpseclib/Net/SSH2.php

```diff
@@ -1546,7 +1552,11 @@ private function key_exchange($kexinit_payload_server = false)
 
             $kexinit_payload_server = $this->get_binary_packet();
 
-            if (!strlen($kexinit_payload_server) || ord($kexinit_payload_server[0]) != NET_SSH2_MSG_KEXINIT) {
+            if (
+                \is_bool($kexinit_payload_server)
+                || !strlen($kexinit_payload_server)
+                || ord($kexinit_payload_server[0]) != NET_SSH2_MSG_KEXINIT
+            ) {
```

Comment:
I almost wonder if it'd be better to do this:

```
switch (true) {
    case is_bool($kexinit_payload_server):
    case !strlen($kexinit_payload_server):
    case ord($kexinit_payload_server[0]) != NET_SSH2_MSG_KEXINIT:
        // ...
}
```
To each their own but I think that's a little easier to read then `$a || $b || $c || $d`, etc

Reply:
I find switch's very ugly and less readable, so I am only going to do this if you insist.

Reply:
It's fine as is!